resources/js/hebrewparsetrainer.js

```diff
@@ -395,14 +395,69 @@ $(document).ready(function(){
 			}
 		});
 
+		restoreSettingsFromStorage();
+
 		if (typeof reload_on_load != 'undefined' && reload_on_load)
 			reloadVerb();
 	}
 
+	function saveSettingsToStorage() {
+		var stems = $.makeArray($('input[name="stem"]:checked').map(function(){return this.value;})).join();
+		var tenses = $.makeArray($('input[name="tense"]:checked').map(function(){return this.value;})).join();
+		var roots = $.makeArray($('select[name="root"]').val()).join();
+		var generalOptions = $.makeArray($('input[name="general"]:checked').map(function(){return this.value;})).join();
```

Comment:
I'm just wondering what the behavior should be if another option is added in the future (this is most likely with `roots`). If the user has selected only a couple of roots (only strong, for example), it makes sense not to include new options (as this implementation does). But if the user has only deselected a couple of roots (only geminates, for instance), or has selected all roots, it makes sense to include new options.

It is not possible to determine what the user had in mind when selecting a number of roots, so this problem cannot really be solved. But perhaps it is possible to add something for the common case where all options are selected? In this case rather than storing the names of all roots we could store e.g. `all`?

There is no need to do this for any options except `roots`.

Reply:
Yes, good thinking. I've handled this by using the empty string, since it defaults to selecting all roots.

Reply:
Thanks!